src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -376,15 +376,16 @@ void writePerPackageRDotJava(
         } else if (useOldStyleableFormat) { // NOPMD  more readable this way, IMO.
           // Nothing extra to do in this case.
 
-        } else if (resourceToIdValuesMap.get(resource) != null) {
-          resource = resource.copyWithNewIdValue(resourceToIdValuesMap.get(resource));
+        } else if (resourceToIdValuesMap.containsKey(resource)) {
+          resource = resourceToIdValuesMap.get(resource);
+          resource = resource.copyWithNewIdValue(resource.idValue);
```

Comment:
nit: don't really like setting resource from the map to resource and then replacing it in the next line
would prefer `resource = resource.copyWithNewIdValue(resourceToIdValuesMap.get(resource).idValue);` or using some local variable